src/Microsoft.AspNet.Hosting/HostingEngine.cs

```diff
@@ -72,7 +80,12 @@ private void EnsureServerFactory(HostingContext context)
                 return;
             }
 
-            context.ServerFactory = _serverManager.LoadServerFactory(context.ServerFactoryLocation);
+            if (_serverLoader == null)
+            {
+                EnsureApplicationServices(context);
```

Comment:
I'm not sure what the pattern is, should we call Ensure before every usage?  or just assume it was already called at the beginning of start?